petastorm/tests/test_parquet_reader.py

```diff
@@ -209,6 +209,33 @@ def preproc_fn1(x):
                (3, 4, 5) == sample['tensor_col_2'].shape[1:]
 
 
+@pytest.mark.parametrize('null_column_dtype', [np.float64, np.unicode_])
+@pytest.mark.parametrize('reader_factory', _D)
+def test_transform_spec_returns_all_none_values(scalar_dataset, null_column_dtype, reader_factory):
+    def fill_id_with_nones(x):
+        return pd.DataFrame({'id': [None] * len(x)})
+
+    edit_fields = [('id', null_column_dtype, (), True)]
+
+    with reader_factory(scalar_dataset.url, schema_fields=["id"],
+                        transform_spec=TransformSpec(fill_id_with_nones, edit_fields=edit_fields)) as reader:
+        sample = next(reader)
+        assert sample.id.dtype.type == null_column_dtype
+
+
+@pytest.mark.parametrize('reader_factory', _D)
+def test_transform_spec_returns_all_none_values_in_a_list_field(scalar_dataset, reader_factory):
+    def fill_id_with_nones(x):
+        return pd.DataFrame({'int_fixed_size_list': [[None for _ in range(3)]] * len(x)})
```

Comment:
We also ran into the NoneType issue with lists of strings. Consider adding string types to the test as well.

The NoneType problem also occurs when only some of the values in the list are None, e.g. `['a', 'b', None]`. What about a `test_transform_spec_returns_some_none_values_in_a_list_field`?